src/main/java/net/hydromatic/streamsqltck/ScriptBuilder.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.hydromatic.streamsqltck;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/** Builder for {@link Script}.
+ *
+ * <p>Example use:
+ *
+ * <blockquote>
+ *     Script s =
+ *       Script.builder()
+ *         .definitions()
+ *         .stream("ORDERS",
+ *           "(TIMESTAMP rowtime, INT orderId, VARCHAR(20) product)")
+ *         .end()
+ *         .query("Q", "select orderId from orders where product = 'milk'")
+ *         .input()
+ *         .insert("ORDERS", 0, 100, "beer")
+ *         .insert("ORDERS", 1, 101, "milk")
+ *         .end()
+ *         .expect()
```

Comment:
This seems not prevent users to have multiple constructs for `expect` and `query`, is this by design?

If yes, what if users specify two "query()" with only one "expect()" in the builder?

Reply:
Maybe we should enforce that a script should has at-least one query and an expectation for each query ?